wapi/util.py

```diff
@@ -182,37 +186,37 @@ def parsetime(datestr, tz=None):
     """
     Parse the input date and optionally convert to correct time zone
     """
+
     d = dateutil.parser.parse(datestr)
+
     if tz is not None:
-        # Convert timestamp to given tz
-        if isinstance(tz, str):
+        if not isinstance(tz, tzinfo):
+            # Raises `pytz.exception.UnknownTimeZoneError`
+            # if `tz` is not the name of a timezone
             tz = pytz.timezone(tz)
-        try:
+
+        if d.tzinfo:
             d = d.astimezone(tz)
-        except ValueError:
-            # This means that d does not already have zone info,
-            # assume the supplied tz is the right one
-            d = d.replace(tzinfo=tz)
+        else:
+            d = tz.localize(d)
+
     else:
         # If datestr does not have tzinfo and no tz given, assume CET
         if d.tzinfo is None:
-            d = d.replace(tzinfo=pytz.timezone('CET'))
+            d = pytz.timezone('CET').localize(d)
     return d
 
 
 def parserange(rangeobj, tz=None):
     """
     Parse a range object (a pair of date strings, which may each be None)
     """
-    try:
-        begin, end = rangeobj
-        if begin is not None:
-            begin = parsetime(begin)
-        if end is not None:
-            end = parsetime(end)
-        return (begin, end)
-    except:
-        raise ValueError('Malformed range: {}'.format(rangeobj))
```

Comment:
Hvorfor ta bort try/except her?

Reply:
Jeg skjønte ikke poenget med å gjøre feilmeldingene mindre spesifikke.

Reply:
Tanken var vel i sin tid å gjøre det enklere for brukerne (som jo gjerne er relativt nye til python), samt spesifikt å passe på at innholdet av `rangeobj` ble med, slik at man fikk se 'root cause'.

Reply:
Forresten, parserange er vel aldri blitt testet, for det er _ikke_ slik en Range serialiseres av WAPI...  Så denne må egentlig skrives helt om, det kommer ikke inn en liste av to timestamps, det som kommer inn er en dictionary med begin/end eller "empty".

Reply:
Vil du jeg skal se på det her, eller burde Noen™ åpne en ny PR på det?

Reply:
Det kan sikkert gå i en ny PR, det er flere ting vi har å fikse på nå.

Reply:
>Tanken var vel i sin tid å gjøre det enklere for brukerne (som jo gjerne er relativt nye til python), samt spesifikt å passe på at innholdet av rangeobj ble med, slik at man fikk se 'root cause'.

Spesifikt gir det nå en bedre feilmelding når årsaken er at `tz` ikke er en tidssone, men kanskje feilmeldingen ble dårligere ellers.

Reply:
Vi får tenke på hvordan gjøre ting riktig når vi fikser tingene etterpå.